docs: Add guide on CA types in Teleport (#43913)
gravitational/teleport: docs/pages/architecture/ca-types.mdx

(i.e. the PostgreSQL server running on a host presents a cert from the `db` CA to a connecting Teleport
`db_service` agent)

Documentation references:
- [Database CA rotations](../management/operations/db-ca-migrations.mdx)

### `db_client`

Issues short-lived certificates for connecting database clients should trust (for exmaple `psql` run by a
user connecting to Teleport's database proxy sees a cert issued by the `db_client` CA)

Documentation references:
- [Database CA rotations](../management/operations/db-ca-migrations.mdx)

### `openssh`

The public key of this CA is installed onto OpenSSH hosts as `TrustedUserCAKeys` to trust users connecting
from Teleport, then short-lived certs are issued from this CA whenever a user makes a connection through
the Teleport proxy to a joined OpenSSH host.

Documentation references:
- [Using Teleport with OpenSSH in agentless mode](../server-access/openssh/openssh.mdx)
- [Using Teleport with OpenSSH in agentless mode (manual installation)](../server-access/openssh/openssh-manual-install.mdx)

### `jwt`

Used to sign and issue certs for JWTs which are automatically sent in headers with requests from

> Teleport's app_service to downstream applications, this CA provides the JWT public key available

Reviewer: ```suggestion
Teleport's app_service to downstream applications. The public keys of this CA are available at https://proxy-address/.well-known/jwks.json
```
